src/components/NavBar/NavigationBar.tsx

```diff
@@ -0,0 +1,188 @@
+    width: 8rem;
+
+    @media (min-width: 800px) {
+        display: none;
+    }
+`;
+
+const LgNavElement = styled.div`
+    display: none;
+    flex-grow: 1;
+    justify-content: end;
+    gap: 1rem;
+    align-items: center;
+
+    @media (min-width: 800px) {
+        display: flex;
+    }
+`;
+
+const Spacer = styled.div`
+    flex-grow: 1;
+`;
+
+const DrawerInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: start;
+    align-items: stretch;
+    padding: 1rem;
+    width: 15rem;
+`;
+
+const LogoNavElement: React.FC<{}> = () => (
+    <UnstyledLink href="/">
+        <LogoDiv>
+            <Logo alt="Vauxhall Foodbank Logo" src="/logo.webp" />
+        </LogoDiv>
+    </UnstyledLink>
+);
+
+const Gap = styled.div`
+    width: 1rem;
+`;
+
+const DrawerButtonWrapper = styled.div`
+    padding: 1rem;
+    border-bottom: 1px solid ${(props) => props.theme.secondaryBackgroundColor};
+
+    &:last-child {
+        border-bottom: none;
+    }
+`;
+
+const DrawerButton = styled(Button)`
+    width: 100%;
+    color: ${(props) => props.theme.secondaryBackgroundColor};
+`;
+
+const StickyAppBar = styled(AppBar)`
+    position: sticky;
+    `;
+
+const ResponsiveAppBar: React.FC<{}> = () => {
+    const [drawer, setDrawer] = React.useState(false);
+
+    const openDrawer = (): void => {
+        setDrawer(true);
+    };
+
+    const closeDrawer = (): void => {
+        setDrawer(false);
+    };
+
+    const pages = [
+        ["Clients", "/clients"],
+        ["Lists", "/lists"],
+        ["Calendar", "/calendar"],
+    ];
+
+    const setThemeMode = useContext(ThemeUpdateContext);
+
+    const switchThemeMode = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setThemeMode(event.target.checked);
+    };
+
+    return (
+        <>
+            <SwipeableDrawer open={drawer} onClose={closeDrawer} onOpen={openDrawer}>
+                <DrawerInner>
+                    {pages.map(([page, link]) => (
+                        <DrawerButtonWrapper key={page}>
+                            <UnstyledLink href={link} onClick={closeDrawer}>
+                                <DrawerButton variant="text">{page}</DrawerButton>
+                            </UnstyledLink>
+                        </DrawerButtonWrapper>
+                    ))}
+                </DrawerInner>
+            </SwipeableDrawer>
+            <StickyAppBar position="static">
+                <AppBarInner>
+                    <SmallNavBarElement>
+                        <IconButton
+                            size="medium"
+                            aria-label="Mobile Menu Button"
+                            onClick={openDrawer}
+                        >
+                            <MenuIcon />
+                        </IconButton>
+                        <Spacer />
+                    </SmallNavBarElement>
+                    <LogoNavElement />
+                    {/* TODO: VFB-16 need to fix nav colours */}
```

Comment:
why is this comment here? what about the colours needs fixing in this section?
(may be best to move this to where the colours are defined)

(also no colons in the TODOs please!)